app/models/shop.rb

```diff
@@ -2,6 +2,10 @@ class Shop < ActiveRecord::Base
   belongs_to :user
   has_many :products, :dependent => :delete_all
 
+  validates :name, length: {minimum: 2, maximum: 15}, presence:true
+  validates :description, length: {minimum: 2}, presence:true
+  validates :email_id, presence:true
```

Comment:
So it does validate that it's a valid email address? so if you enter just `aksjld8129` there is an error?